AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.h

```diff
@@ -16,14 +18,33 @@ NS_ASSUME_NONNULL_BEGIN
 + (void)trackEvent:(NSString *)eventName;
 
 /**
- * Track an event.
+ * Track an event with properties.
  *
  * @param eventName  event name.
  * @param properties dictionary of properties.
  */
 + (void)trackEvent:(NSString *)eventName
     withProperties:(nullable NSDictionary<NSString *, NSString *> *)properties;
 
+/**
+ * Track an event with typed properties.
+ *
+ * @param eventName  event name.
+ * @param properties an MSEventProperties object.
+ *
+ * @discussion For events going to App Center, the following validation rules are applied:
```

Comment:
Do you want to mention validation rules for OC?

Reply:
I think validation rules in the API doc would be awesome!

Reply:
We thought that we would leave them out for this PR, since it is unrelated to one collector. (Whoever does the OC work will probably benefit by adding the rules themselves, as it's a useful sanity check when you write them in)